src/main/java/com/monumental/services/ModelService.java

```diff
@@ -397,4 +421,62 @@ private void initializeLazyLoadedCollections(List<T> records) {
             }
         }
     }
+
+    /**
+     * TODO: Delete and replace with JPA 2 code
+     */
+    Session openSession() {
+        return this.sessionFactoryService.getFactory().openSession();
+    }
+
+    /**
+     * This is handles your connection to the database and has the CriteriaBuilder reference
+     */
+    public EntityManager getEntityManager() {
```

Comment:
Any reason this is in a function and we're not just calling `this.entityManager.blah`?

Reply:
I don't think you can make Spring beans public. Also, muh getters and setters. Technically this makes it readonly to the public